tests/Smoke/InstallViaComposerTest.php

```diff
@@ -164,8 +157,55 @@ public function testInstallationViaArtifactIsPossible(): void
         self::assertCommandsWork($stepsToPrepareArtifact, $tmpArtifactPath);
         self::assertCommandsWork($this->stepsToVerifyInstallation, $tmpPath);
 
-        $fs->remove($tmpPath);
-        $fs->remove($tmpArtifactPath);
+        $this->fs->remove($tmpPath);
+        $this->fs->remove($tmpArtifactPath);
+    }
+
+    public function testDoctrineAnnotationRulesetThrowsAnExceptionWhenDoctrinePackagesAreNotInstalled(): void
+    {
+        $tmpPath = $this->createFakeComposerProject($this->currentCodeAsComposerDependency);
+
+        self::assertCommandsWork(
+            [
+                'composer install -q',
+                'composer dump-autoload --optimize',
+                'php vendor/autoload.php',
+                'echo "<?php class /** @SomeAnnotation() */Foo {}" > test.php',
+            ],
+            $tmpPath
+        );
+
+        $this->expectException(ExecutionException::class);
+        $this->expectExceptionMessageMatches('|.*You need to install `doctrine/annotations` and `doctrine/lexer` to be able to use.*|');
+
+        CommandExecutor::create('vendor/bin/php-cs-fixer fix --dry-run -vvv --rules=@DoctrineAnnotation test.php', $tmpPath)->getResult();
+
+        $this->fs->remove($tmpPath);
+    }
+
+    public function testDoctrineAnnotationRulesetWorksIfSuggestedDependenciesAreInstalled(): void
+    {
+        $tmpPath = $this->createFakeComposerProject($this->currentCodeAsComposerDependency);
+
+        self::assertCommandsWork(
+            [
+                'composer install -q',
+                'composer require doctrine/annotations doctrine/lexer',
+                'composer dump-autoload --optimize',
+                'php vendor/autoload.php',
+                'echo "<?php class /** @SomeAnnotation() */Foo {}" > test.php',
+            ],
+            $tmpPath
+        );
+
+        self::assertSame(
+            0,
+            CommandExecutor::create('vendor/bin/php-cs-fixer fix --rules=@DoctrineAnnotation test.php', $tmpPath)
```

Comment:
```suggestion
            CommandExecutor::create('vendor/bin/php-cs-fixer fix --dry-run -vvv --rules=@DoctrineAnnotation test.php', $tmpPath)
```
Aligning to the same as test above.